File: projects/08/ruby/TempTester.rb
# testTemp.rb tests the module Temperature.rb
#
# Created by: Sarah Whitten
# Date: April 20, 2019

require './Temperature.rb'

def testTemp
	# create the baseTemp and limitTemp with "default" values
   baseTemp = Temperature::TemperatureClass.new(0, 'K')
   limitTemp = Temperature::TemperatureClass.new(0, 'K')
   # ask the user for input and change the temperatures accordingly
   puts "Base temperature:"
   baseTemp.input
   puts "Limit temperature:"
   limitTemp.input
   # get the increment value
   puts "Increment value:"
   stepValue = gets.to_f
   # while the baseTemp is less than or equal to the limitTemp (regardless of scale)
   # continues looping, printing the temperature and incrementing it
   while (baseTemp.lessthan(baseTemp, limitTemp) or baseTemp.equals(baseTemp, limitTemp))
      baseTemp.print
      baseTemp = baseTemp.raise(baseTemp, stepValue)
   end
end

testTemp

File: projects/08/java/TempTester.java
// TempTester.java tests the methods from the
// Temperature.java class
// 
// Created by: Sarah Whitten
// Date: April 20, 2019

import java.util.Scanner;

public class TempTester
{
	public static void main(String[] args)
	{
		Scanner keyboard = new Scanner(System.in);
		System.out.println("Creating the lower temperature");
		Temperature baseTemp = new Temperature(0, 'C');
		baseTemp.readTemp();
		System.out.println("Creating the upper temperature");
		Temperature limitTemp = new Temperature(0, 'C');
		limitTemp.readTemp();
		System.out.print("Please enter the step value: ");
		double stepValue = keyboard.nextDouble();

		while(Temperature.lessthan(baseTemp, limitTemp) || Temperature.equals(baseTemp, limitTemp))
		{
			System.out.println(baseTemp);
			baseTemp = Temperature.raise(baseTemp, stepValue);
		}
	}
}

